py/torch_tensorrt/dynamo/debug/_Debugger.py

```diff
@@ -0,0 +1,227 @@
+
+    def __enter__(self) -> None:
+        self.original_lvl = _LOGGER.getEffectiveLevel()
+        self.rt_level = torch.ops.tensorrt.get_logging_level()
+        dictConfig(self.get_customized_logging_config())
+
+        if self.capture_fx_graph_before or self.capture_fx_graph_after:
+            self.old_pre_passes, self.old_post_passes = (
+                ATEN_PRE_LOWERING_PASSES.passes,
+                ATEN_POST_LOWERING_PASSES.passes,
+            )
+            pre_pass_names = [p.__name__ for p in self.old_pre_passes]
+            post_pass_names = [p.__name__ for p in self.old_post_passes]
+            path = os.path.join(self.cfg.logging_dir, "lowering_passes_visualization")
+            if self.capture_fx_graph_before is not None:
+                pre_vis_passes = [
+                    p for p in self.capture_fx_graph_before if p in pre_pass_names
+                ]
+                post_vis_passes = [
+                    p for p in self.capture_fx_graph_before if p in post_pass_names
+                ]
+                ATEN_PRE_LOWERING_PASSES.insert_debug_pass_before(pre_vis_passes, path)
+                ATEN_POST_LOWERING_PASSES.insert_debug_pass_before(
+                    post_vis_passes, path
+                )
+            if self.capture_fx_graph_after is not None:
+                pre_vis_passes = [
+                    p for p in self.capture_fx_graph_after if p in pre_pass_names
+                ]
+                post_vis_passes = [
+                    p for p in self.capture_fx_graph_after if p in post_pass_names
+                ]
+                ATEN_PRE_LOWERING_PASSES.insert_debug_pass_after(pre_vis_passes, path)
+                ATEN_POST_LOWERING_PASSES.insert_debug_pass_after(post_vis_passes, path)
+
+        self._context_stack = contextlib.ExitStack()
+
+        for f in _DEBUG_ENABLED_FUNCS:
+            f.__kwdefaults__["_debugger_settings"] = self.cfg
+
+        [
+            self._context_stack.enter_context(
+                mock.patch.object(
+                    c,
+                    "__init__",
+                    functools.partialmethod(c.__init__, _debugger_settings=self.cfg),
+                )
+            )
+            for c in _DEBUG_ENABLED_CLS
+        ]
+
+    def __exit__(self, exc_type: Any, exc_value: Any, exc_tb: Any) -> None:
+
+        dictConfig(self.get_default_logging_config())
+        torch.ops.tensorrt.set_logging_level(self.rt_level)
+        if self.capture_fx_graph_before or self.capture_fx_graph_after:
+            ATEN_PRE_LOWERING_PASSES.passes, ATEN_POST_LOWERING_PASSES.passes = (
+                self.old_pre_passes,
+                self.old_post_passes,
+            )
+        self.debug_file_dir = tempfile.TemporaryDirectory().name
+
+        for f in _DEBUG_ENABLED_FUNCS:
+            f.__kwdefaults__["_debugger_settings"] = None
+
+        self._context_stack.close()
+
+    def get_customized_logging_config(self) -> dict[str, Any]:
```

Comment:
Consider merging `get_customized_logging_config` and `get_default_logging_config` by using 
```
def get_logging_config(self, log_level): 
```
since most of the code is same b/w them